From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sylvester Kaczmarek
 <16242628+sylvesterkaczmarek@users.noreply.github.com>
Date: Sun, 16 Aug 2026 13:28:42 +0100
Subject: [PATCH] 9508: #9505 Wrap long Array Values entries in the Listing

#9505 Wrap long array values in the Listing
---
 .../util/viewer/field/ArrayValuesFieldFactory.java  | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/viewer/field/ArrayValuesFieldFactory.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/viewer/field/ArrayValuesFieldFactory.java
index c8ec329427..b8ce501edc 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/viewer/field/ArrayValuesFieldFactory.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/viewer/field/ArrayValuesFieldFactory.java
@@ -16,9 +16,12 @@
 package ghidra.app.util.viewer.field;
 
 import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
 
 import docking.widgets.fieldpanel.field.*;
 import docking.widgets.fieldpanel.support.FieldLocation;
+import docking.widgets.fieldpanel.support.FieldUtils;
 import docking.widgets.fieldpanel.support.RowColLocation;
 import ghidra.app.util.ListingHighlightProvider;
 import ghidra.app.util.viewer.format.FieldFormatModel;
@@ -98,17 +101,19 @@ public class ArrayValuesFieldFactory extends FieldFactory {
 		int itemCount = Math.min(remaining, valuesPerLine);
 		boolean isLastLine = remaining <= itemCount;
 
-		FieldElement[] aStrings = new FieldElement[itemCount];
+		List<FieldElement> elements = new ArrayList<>();
 		for (int i = 0; i < itemCount; i++) {
 			Data child = parent.getComponent(index++);
 			boolean isLastItem = isLastLine && (i == itemCount - 1);
 			String value = getDisplayValue(child, !isLastItem);
 			AttributedString as =
 				new AttributedString(value, ListingColors.ARRAY_VALUES, getMetrics());
-			aStrings[i] = new TextFieldElement(as, i, 0);
+			TextFieldElement element = new TextFieldElement(as, i, 0);
+			elements.addAll(FieldUtils.wrap(element, width, false));
 		}
-		return ListingTextField.createPackedTextField(this, proxy, aStrings, startX + varWidth,
-			width, 1, hlProvider);
+		return ListingTextField.createPackedTextField(this, proxy,
+			elements.toArray(new FieldElement[0]), startX + varWidth, width, Integer.MAX_VALUE,
+			hlProvider);
 
 	}
 
-- 
2.45.1

